dask/array/linalg.py

```diff
@@ -429,8 +429,7 @@ def qr(a, name=None):
     dask.array.linalg.sfqr: Implementation for short-and-fat arrays
     """
 
-    # if len(a.chunks[1]) == 1 and a.shape[0] >= a.shape[1]:
-    if len(a.chunks[1]) == 1: # and a.shape[0] >= a.shape[1]:
+    if len(a.chunks[1]) == 1 and len(a.chunks[0]) > 1:
```

Comment:
Is this always the right choice?

Reply:
Yes. In the one block case, `tsqr` does too much work (two decompositions). `sfqr` is more economical with exactly one.